### PR TITLE
Async support

### PR DESCRIPTION
**Pull request by ridomin**
This pull request changes the `ICommand` interface to:

```cs
public interface ICommand
    {
        Task<int> Execute();
    }
```

It's a breaking change with existing clients, but the required changes are minimal. I've added basic tests.

### CODE REVIEW

File: src/examples/JustCli.TestAppNetCore/JustCli.TestAppNetCore.csproj [hunk 1]
```diff
@@ -6,6 +6,10 @@
     <Configurations>Debug;Release;ReleaseForPublish</Configurations>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <LangVersion>latest</LangVersion>
```
Comment (ridomin) [inline src/examples/JustCli.TestAppNetCore/JustCli.TestAppNetCore.csproj]: This is needed to support async Main methods in Console apps

**Comment by rido-min** [comment]
Any change to merge this PR soon?

**Comment by alexpantyukhin** [comment]
I think the idea of the PR is very useful. It makes ability to use `await` for async operations without `Wait()` on all async calls.

But for my opinion we should support two types of command: `ICommand` and `ICommandAsync`. Actually if it's a new tool there is no need to support not async commands. But the lib could be useful by other consumers, and I guess the backward compatibility should be supported.

**Comment by jden123** [comment]
@ridomin Thank you for your contribution. And sorry for late reply.
That's good idea! I'm thinking how to prevent breaking changes.

**Comment by jden123** [comment]
I added async support. Thank you.

**Pull request closed without merging**